crates/client/eth/src/state_update.rs

```diff
@@ -0,0 +1,261 @@
+    #[tokio::test]
+    async fn test_event_subscription() {
+        // Set up chain info
+        let chain_info =
+            ChainInfo { chain_id: Felt::from_dec_str("11153111").unwrap(), chain_name: "testing".to_string() };
+
+        // Set up database paths
+        let current_dir = env::current_dir().expect("Failed to get current directory");
+        let base_path = current_dir.join("data");
+        let backup_dir = Some(current_dir.parent().expect("Failed to get parent directory").join("backups"));
+
+        // Initialize database service
+        let db = Arc::new(
+            DatabaseService::new(&base_path, backup_dir, false, &chain_info)
+                .await
+                .expect("Failed to create database service"),
+        );
+
+        // Set up metrics service
+        let prometheus_service = MetricsService::new(true, false, 9615).unwrap();
+        let block_metrics = BlockMetrics::register(&prometheus_service.registry()).unwrap();
+
+        // Set up Anvil for local Ethereum testing
+        let anvil = Anvil::new()
+            .fork(ETH_URL)
+            .fork_block_number(L1_BLOCK_NUMBER)
+            .try_spawn()
+            .expect("Failed to spawn Anvil instance");
+        let rpc_url: Url = anvil.endpoint().parse().expect("Failed to parse RPC URL");
+
+        // Set up Ethereum provider and contracts
+        let provider = ProviderBuilder::new().on_http(rpc_url.clone());
+        let contract = DummyContract::deploy(provider.clone()).await.unwrap();
+        let core_contract = StarknetCoreContract::new(contract.address().clone(), provider.clone());
+
+        let eth_client = EthereumClient { provider: Arc::new(provider), l1_core_contract: core_contract.clone() };
+
+        // Set up stop flag for graceful shutdown
+        let stop_flag = Arc::new(AtomicBool::new(false));
+        let stop_flag_clone = stop_flag.clone();
+
+        // Start listening for state updates
+        let listen_handle = {
+            let db = Arc::clone(&db);
+            tokio::spawn(async move {
+                listen_and_update_state(eth_client, &db.backend(), block_metrics.clone(), chain_info.chain_id).await
+            })
+        };
+
+        // Function to fire events
+        async fn fire_event(contract: &DummyContractInstance<Http<Client>, RootProvider<Http<Client>>>) {
+            let _ = contract.fireEvent().send().await.expect("Failed to fire event");
+        }
+
+        // Fire a single event
+        let event_handle = tokio::spawn({
+            let stop_flag = stop_flag_clone;
+            async move {
+                if !stop_flag.load(Ordering::Relaxed) {
+                    fire_event(&contract).await;
+                }
+            }
+        });
+
+        // Wait for event processing
+        tokio::time::sleep(Duration::from_secs(EVENT_PROCESSING_TIME)).await;
+
+        // Signal to stop event firing
+        stop_flag.store(true, Ordering::Relaxed);
```

Comment:
I think we can synchronously wait for an event and just check the DB straight away

Reply:
yes removed this